### ISSUE TITLE
MariaDB IP address instead of hostname in config file

### ISSUE THREAD

**Issue by pinkvoid** [bitnami/bitnami-docker-phpbb#48]
Hi,
I've just tried the lastest docker image and after restarting the whole service (PhpBB and MariaDB) I got an mysql connection refused error.

I've found out that in PhpBB's config file the IP address of the mariadb container was set instead of the hostname. Plus, after a restart, the IP addresses change.

Replacing the IP address with the hostname 'mariadb' fixed this issue for me.

It seems like the hostname is always resolved when creating the service, at least with Docker Composer.

**Comment by dbarranco** [comment]
Hi @pinkvoid 

I am sorry for the delay on my answer.
Could you please answer the following questions?

How did you restarted the services?
Did you deleted all the volumes while performing the restart?
Did you modify the docker-compose file?

The database host it's set in the [Dockerfile](https://github.com/bitnami/bitnami-docker-phpbb/blob/master/3/Dockerfile), so a restart shouldn't affect to the behaviour.

Kind regards!

**Comment by ericblade** [comment]
I don't mean to be cold, but it probably needs to be pointed out that if you can't reproduce it, then you don't understand any part of the report, don't understand how phpbb works, and likely didn't attempt to reproduce it.

phpbb's configuration in this image is stored in /bitnami/phpbb/config.php .
After running the initial configuration step, and starting the board, something writes the host IP address to that file, instead of the host DNS name.

This prevents the container from being restarted, as when it restarts, it is now attempting to access the database specified in the config.php file, instead of the one specified in the Dockerfile.  The entire contents of the stock config.php in regards to database access are incorrect, after the initial configuration, so restarting the phpbb service in any fashion, will fail until that file is corrected.

The part that I don't understand, without digging in much further than I have time to, is how it even manages to work on the first run.  

docker-compose stop, docker-compose down, docker-compose restart, docker stop, etc, will all trigger the failure, which appears to be root caused because the contents of phpbb.conf are incorrect.

The questions that you ask are relevant to the symptom, but are not at all relevant to the bug.  The bug is either that phpbb.conf is being written incorrectly, or it is not being overridden correctly by the environment variables, or both.

**Comment by sameersbn** [comment]
@dbarranco I too was unable to reproduce the issue after several tries. In any case, it's clear that users can face this issue if the ip address of the mariadb container changes after a restart and as  @pinkvoid pointed out we should be using the database hostname in the config instead of using the resolved ip address which would make it resilient to ip address changes.

> despite the persistent data being stored in a volume, the data on the volumes gets erased when the services are brought back up.

@ericblade We do not expect this to happen. Can you please create a separate issue and provide more details regarding the environment and setup. Use the issue template to post the required information.

**Comment by ericblade** [comment]
To repro, just do docker-compose up -d, login and verify it works, then docker-compose restart phpbb, and it won't come up.  At least not in Windows. I don't know if maybe there's something different going on in the Linux hosts that makes that IP address work, but it doesn't from my Windows host.

So, yes, the correct host needs to be written to the conf, and/or the environment override to that needs to work (since i can clearly see it being set to "mariadb" in the Dockerfile..)

will try to repro that issue again later and file it.  i was trying to get a volume mounted for the extensions, and that's when i discovered both of these problems -- once i got my phpbb image working again, all the data had been reset.

**Comment by dbarranco** [comment]
Hi @ericblade @pinkvoid 

I'm sorry for the delay on my answer.
I saw the configuration file of phpBB and it has configured the database with the ip address of the container and not the hostname:

```
<?php
// phpBB 3.2.x auto-generated configuration file
// Do not change anything in this file!
$dbms = 'phpbb\\db\\driver\\mysqli';
$dbhost = '172.26.0.2';
$dbport = '3306';
$dbname = 'bitnami_phpbb';
$dbuser = 'bn_phpbb';
$dbpasswd = 'examplePasswd';
$table_prefix = 'phpbb_';
$phpbb_adm_relative_path = 'adm/';
$acm_type = 'phpbb\\cache\\driver\\file';
```

I have created an internal ticket to solve this and also re-check that we persist all the data of the application. We are currently persisting all the files that has a symbolic link:

```
root@98833e1bb8ef:/opt/bitnami/phpbb# ls -la
total 600
drwxr-xr-x  1 root   root     4096 Nov 14 08:17 .
drwxr-xr-x  1 root   root     4096 Aug 22 16:08 ..
-rw-r--r--  1 root   root        0 Jul 17 00:21 .buildcomplete
-rw-r-----  1 root   root      148 Nov 14 08:16 .htaccess
drwxr-xr-x  4 root   root     4096 Aug 22 16:08 adm
-rw-r--r--  1 root   root      952 Jul 17 00:21 app.php
drwxr-xr-x  7 root   root     4096 Aug 22 16:08 assets
drwxr-xr-x  2 root   root     4096 Aug 22 16:08 bin
drwxr-xr-x  1 daemon daemon   4096 Nov 14 08:17 cache
-rw-r--r--  1 root   root     5315 Jul 17 00:21 common.php
-rw-r--r--  1 root   root     1955 Jul 17 00:21 composer.json
-rw-r--r--  1 root   root   126856 Jul 17 00:21 composer.lock
drwxr-xr-x  1 root   root     4096 Aug 22 16:08 config
lrwxrwxrwx  1 root   root       25 Nov 14 08:17 config.php -> /bitnami/phpbb/config.php
-rw-r--r--  1 root   root     2091 Jul 17 00:21 cron.php
drwxr-xr-x  3 root   root     4096 Aug 22 16:08 docs
drwxr-xr-x  2 root   root     4096 Aug 22 16:08 download
drwxr-xr-x  3 root   root     4096 Aug 22 16:08 ext
-rw-r--r--  1 root   root      944 Jul 17 00:21 faq.php
-rw-r--r--  1 root   root     1646 Jul 17 00:21 feed.php
lrwxrwxrwx  1 root   root       20 Nov 14 08:17 files -> /bitnami/phpbb/files
lrwxrwxrwx  1 root   root       21 Nov 14 08:17 images -> /bitnami/phpbb/images
drwxr-xr-x  1 root   root     4096 Aug 22 16:08 includes
-rw-r--r--  1 root   root     8294 Jul 17 00:21 index.php
drwxr-xr-x  3 root   root     4096 Aug 22 16:08 language
drwxr-xr-x  2 root   root     4096 Aug 22 16:08 licenses
-rw-r--r--  1 root   root     9178 Jul 17 00:21 mcp.php
-rw-r--r--  1 root   root    54377 Jul 17 00:21 memberlist.php
drwxr-xr-x  1 root   root     4096 Aug 22 16:08 phpbb
-rw-r--r--  1 root   root    74337 Jul 17 00:21 posting.php
-rw-r--r--  1 root   root     1097 Jul 17 00:21 report.php
-rw-r--r--  1 root   root    55770 Jul 17 00:21 search.php
lrwxrwxrwx  1 root   root       20 Nov 14 08:17 store -> /bitnami/phpbb/store
drwxr-xr-x  4 root   root     4096 Aug 22 16:08 styles
-rw-r--r--  1 root   root    11602 Jul 17 00:21 ucp.php
drwxr-xr-x 18 root   root     4096 Aug 22 16:08 vendor
-rw-r--r--  1 root   root    39019 Jul 17 00:21 viewforum.php
-rw-r--r--  1 root   root    16942 Jul 17 00:21 viewonline.php
-rw-r--r--  1 root   root    89084 Jul 17 00:21 viewtopic.php
-rw-r--r--  1 root   root     1177 Jul 17 00:21 web.config
```

In the meantime. You can change the database hostname in the `config.php` file and for next restarts it will keep the information, so you won't see again the error when the container starts.
If you need to persist any other directory (as any folder for extensions) you can create a symbolic link with that:

```
ln -s full/path/to/folder /bitnami/phpbb/
```

For any next restart of the container do not forger that you will have to restore again that symbolic link. This is just a temporal fix in the mean time we evaluate all the directories that can be persisted and we include that by default in the container.

Kind regards

**Comment by tompizmor** [comment]
Hi @pinkvoid , @ericblade 

I just wanted to mention that this issue should be already fixed in the latest release (Docker tags `3.2.2-ol-7-r27` and `3.2.2-debian-9-r23`)

Regards,
Tomas
